docroot/profiles/custom/cgov_site/modules/custom/cgov_site_section/cgov_site_section.module

```diff
@@ -119,35 +164,178 @@ function cgov_site_section_taxonomy_term_update(EntityInterface $entity) {
+}
 
+/**
+ * Invalidate the cache tag equal to the given Site Sections ID.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $site_section_entity
+ *   The Site Section to invalidate.
+ */
+function invalidate_section_cache_tag(EntityInterface $site_section_entity) {
+  $section_id = $site_section_entity->id();
+  // Get the sections ID and invalidate the cache tag.
+  if ($section_id) {
+    Cache::invalidateTags(['site_section:' . $section_id]);
+  }
+}
+
+/**
+ * Saves the children of the given Site Section taxonomy term.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $updated_section_entity
+ *   The site section entity with its pending value updates.
+ */
+function save_term_children(EntityInterface $updated_section_entity) {
+  $term_storage = \Drupal::service('entity_type.manager')
+    ->getStorage("taxonomy_term");
   // Update immediate child sections.
-  $subsections = $term_storage->loadChildren($entity->id(), 'cgov_site_sections');
+  $subsections = $term_storage->loadChildren($updated_section_entity->id(), 'cgov_site_sections');
 
-  // Resave the sections. This should trigger the presave above with will set
+  // Re-save the sections. This should trigger the presave above with will set
   // the computed_path to the new path.
   foreach ($subsections as $section) {
     $section->save();
   }
+}
 
-  // TODO: Figure out how to get pathauto to update all the aliases using
-  // pathauto. To do it here is A) too slow, and B) fraught with
-  // issue regarding workflow and the various entity types we would have to hit.
+/**
+ * Determines if a section should trigger a re-save of its children terms.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $updated_section_entity
+ *   The site section entity with its pending value updates.
+ *
+ * @return bool
+ *   A TRUE or FALSE as to whether to trigger a re-save.
+ */
+function will_trigger_save_children(EntityInterface $updated_section_entity) {
+
+  $fields = [
+    'computed_path',
+    'parent',
+    'field_pretty_url',
+    'langcode',
+  ];
+  return entity_fields_modified($fields, $updated_section_entity);
+}
+
+/**
+ * Determine if an entity should trigger a cache tag invalidation.
+ *
+ * Fields which trigger a cache tag invalidation for a section when modified:
+ *  name
+ *  field_landing_page
+ *  field_pretty_url
+ *  field_section_nav_root
+ *  field_navigation_label
+ *  field_navigation_display_options
+ *  weight.
+ *
+ * For a node or media item:
+ * field_site_section
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $updated_entity
+ *   The entity just updated.
+ *
+ * @return bool
+ *   A TRUE or FALSE as to whether to trigger a cache clear.
+ */
+function _will_trigger_cache_tag_invalidation(EntityInterface $updated_entity) {
+  $values = [];
+
+  // Invalidation Check For Nodes/Media.
+  if ($updated_entity->getEntityTypeId() === 'node' || $updated_entity->getEntityTypeId() === 'media') {
```

Comment:
Whether or not a node/media should invaildate it's term taxonomy.

Reply:
This should also check pretty_url_name since that makes up the HREF in the nav.

Reply:
Does it really though? This is where I got thrown a bit. 

If a page is in the Section Nav, doesn't that mean it's always a landing page? And landing pages do not set a pretty URL on the node/media , their path is derived from the Site Section pretty URL (via the computed path field). 

Is there a scenario where a page is set as a landing page but doesn't have an empty pretty URL field on the node/media?

Reply:
I see what you are getting at, but technically any node can go in as a landing page. Which means that whatever makes up its URL needs to be checked for changes. (Also, Media does not need to be checked because only nodes can be landing pages.)